code/modules/mob/living/living_movement.dm

```diff
@@ -1,7 +1,53 @@
 /mob/living/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
 	. = ..()
 	update_turf_movespeed(loc)
+	if(HAS_TRAIT(src, TRAIT_NEGATES_GRAVITY))
```

Comment:
Moving this logic to a zero-gravity element for groundless turfs `Entered` and `Exited` procs would be more ideal.  
 `Moved` is getting spammed by every player, every second, all round.  Most people won't even interact with zero-gravity turfs on an average round.

Reply:
This isn't what handles space nograv, that's caused by the force_nograv var later on. That said:
an element is not a workable solution due to the scale of space turfs. any extra cost to creating them is a huge nutshot to init, and elements aren't free. 
doing it hardcoded off entered/exited breaks the promise of the var, and makes things more locked in which I don't want to do. It also potentially leads to grav refresh spam which I don't want. It ends up being a cost of at most like 0.1/0.2 seconds in the worst possible round, and I think that's acceptable.